Tests/Algorithm/Framework/Selection/OpenInterestFutureUniverseSelectionModelTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Moq;
+using NodaTime;
+using NUnit.Framework;
+using QuantConnect.Algorithm.Farmework.Selection;
+using QuantConnect.Configuration;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Tests.Algorithm.Framework.Selection
+{
+    [TestFixture]
+    public class OpenInterestFutureUniverseSelectionModelTests
+    {
+        private const int TestChainContractLookupLimit = 6;
+        private const int TestResultsLimit = 4;
+        private static readonly Symbol Jan = Symbol.CreateFuture(Futures.Metals.Gold, Market.COMEX, new DateTime(2020, 01, 01));
+        private static readonly Symbol Feb = Symbol.CreateFuture(Futures.Metals.Gold, Market.COMEX, new DateTime(2020, 02, 01));
+        private static readonly Symbol March = Symbol.CreateFuture(Futures.Metals.Gold, Market.COMEX, new DateTime(2020, 03, 01));
+        private static readonly Symbol April = Symbol.CreateFuture(Futures.Metals.Gold, Market.COMEX, new DateTime(2020, 04, 01));
+        private static readonly DateTime TestDate = new DateTime(2020, 05, 11);
+        private static readonly DateTime ExpectedPreviousDate = new DateTime(2020, 05, 08);
+        private static readonly IReadOnlyDictionary<Symbol, decimal> OpenInterestData = new Dictionary<Symbol, decimal>
+        {
+            [Jan] = 3,
+            [Feb] = 6,
+            [March] = 3, // Same as Jan.
+            [April] = 1
+        };
+        private Mock<IHistoryProvider> _mockHistoryProvider;
+        private OpenInterestFutureUniverseSelectionModel _underTest;
+
+        [SetUp]
+        public void SetUp()
+        {
+            Config.Reset();
+            MarketHoursDatabase.Reset();
+            Config.Set("data-folder", Path.Combine(TestContext.CurrentContext.TestDirectory, "..", "..", "..", "Data"));
```

Comment:
Wondering why are these required, I gave it a quick test and it seemed to work correctly without them too?
```
            Config.Reset();
            MarketHoursDatabase.Reset();
            Config.Set("data-folder", Path.Combine(TestContext.CurrentContext.TestDirectory, "..", "..", "..", "Data"));
```

Reply:
Left-over from code in our own project. :P